cli/include/hse/cli/rest/api.h

```diff
@@ -27,7 +27,7 @@ merr_t
 rest_kvdb_cancel_compaction(const char *alias);
 
 merr_t
-rest_kvdb_compact(const char *alias);
+rest_kvdb_compact(const char *alias, bool full);
```

Comment:
Seems like the rest changes are unnecessary since we said this should only be done on an offline KVDB for now. Thoughts?

Reply:
Maybe offline KVDB is the wrong terminology here. Maybe the better way to think about it is that the CLI should own the KVDB when a full compaction is requested.

Reply:
The CLI compaction code uses REST even when the CLI had to open the KVDB.

Reply:
Hmm is there something I am missing as to why this would be the case?

Reply:
Less code to maintain.  This code does this:
```
if (not already open)
    open kvdb;
use REST;
if (i opened kvdb)
    close kvdb;
```
vs this:
```
if (not already open) {
    open kvdb;
    use C API;
    close KVDB;
} else {
    use kvdb;
}
```

Reply:
Oh I guess it is to just have one path. Makes sense.

Reply:
We had this convo a while back and I thought you were firmly in the camp of "use C API if you can open the KVDB, only use REST if it is already open".